iothub_client/samples/iotedge_downstream_device_sample/iotedge_downstream_device_sample.c

```diff
@@ -0,0 +1,313 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+// CAVEAT: This sample is to demonstrate azure IoT client concepts only and is not a guide design principles or style
+// Checking of return codes and error values shall be omitted for brevity. Please practice sound engineering practices
+// when writing production code.
+
+#include <stdio.h>
+#include <stdlib.h>
+
+#include "iothub.h"
+#include "iothub_device_client.h"
+#include "iothub_client_options.h"
+#include "iothub_message.h"
+#include "azure_c_shared_utility/threadapi.h"
+#include "azure_c_shared_utility/crt_abstractions.h"
+#include "azure_c_shared_utility/platform.h"
+#include "azure_c_shared_utility/shared_util_options.h"
+
+/* This sample uses the convenience APIs of iothub_client for example purposes. */
+
+// The protocol you wish to use should be uncommented
+//
+#define SAMPLE_MQTT
+//#define SAMPLE_MQTT_OVER_WEBSOCKETS
+//#define SAMPLE_AMQP
+//#define SAMPLE_AMQP_OVER_WEBSOCKETS
+//#define SAMPLE_HTTP
+
+#ifdef SAMPLE_MQTT
+    #include "iothubtransportmqtt.h"
+#endif // SAMPLE_MQTT
+#ifdef SAMPLE_MQTT_OVER_WEBSOCKETS
+    #include "iothubtransportmqtt_websockets.h"
+#endif // SAMPLE_MQTT_OVER_WEBSOCKETS
+#ifdef SAMPLE_AMQP
+    #include "iothubtransportamqp.h"
+#endif // SAMPLE_AMQP
+#ifdef SAMPLE_AMQP_OVER_WEBSOCKETS
+    #include "iothubtransportamqp_websockets.h"
+#endif // SAMPLE_AMQP_OVER_WEBSOCKETS
+#ifdef SAMPLE_HTTP
+    #include "iothubtransporthttp.h"
+#endif // SAMPLE_HTTP
+
+#ifdef SET_TRUSTED_CERT_IN_SAMPLES
```

Comment:
>SET_TRUSTED_CERT_IN_SAMPLES [](start = 7, length = 27)

Please remove this #ifdef block.  We don't want these certs being used but instead what comes from the file.

Reply:
Fixed.